src/fcnnls.ts

```diff
@@ -153,27 +169,83 @@ export default function fcnnls(
           }
         }
         m = Hset.length;
+
+        if (info) {
+          errorLog.mseRowVector.push(getRSE({ X, K, Y, error }));
+          ++errorLog.nCalculationsOfK;
+        }
       }
     }
+    if (Hset.length === 0 || (iter === maxIterations && info)) {
+      errorLog.mseRowVector.push(getRSE({ X, K, Y, error }));
+      ++errorLog.nCalculationsOfK;
+    }
 
-    const newParam = optimality(
+    const newParam = optimality({
       iter,
-      maxIterations,
+      maxIter: maxIterations,
       XtX,
       XtY,
       Fset,
       Pset,
       W,
       K,
-      l,
-      p,
+      l: nColsX,
+      p: nColsY,
       D,
       gradientTolerance,
-    );
+    });
     Pset = newParam.Pset;
     Fset = newParam.Fset;
     W = newParam.W;
   }
-
+  if (info) {
+    return { K, errorLog };
+  }
   return K;
 }
+
+export interface FcnnlsOptions {
+  /**
+   * Number of iterations
+   * @default 3 times the number of columns of X
+   */
+  maxIterations?: number;
+  /**
+   * Larger values (like 1e-4) could help if the number of iterations is exceeded. For most cases, lower values should also be fine.
+   * @default 1e-5
+   */
+  gradientTolerance?: number;
+  /**
+   * Whether to return the gradient matrix at every cycle and number of iterations.
+   */
+  info?: boolean;
+  /**
+   * @default true. (The actual value is undefined.) `false` will add a column of ones to the left of X.
+   */
+  interceptAtZero?: boolean;
+}
+export type FcnnlsOutput = OutMatrixK | OutMatrixAndLog;
```

Comment:
The return shape should be the same with `info` (previously `errorLog`) optional depending the options.
Please check how to define the right return type based on the options.info value. This can be done in Typescript with function overloads.